core/src/components/input/input.tsx

```diff
@@ -590,10 +589,14 @@ export class Input implements ComponentInterface {
       return [
         <div class="input-outline-container">
           <div class="input-outline-start"></div>
-          <div class="input-outline-notch">{needsNotch && this.renderLabel()}</div>
+          <div class="input-outline-notch">
```

Comment:
Why do we now render the notch for all label placements, rather than just the ones that need it?

Also, if this is definitely needed, the comment above here should be updated since it still mentions only rendering the notch for certain label placements.

Reply:
The way the outline works with `fill="outline"` is that it creates 3 elements: The start, the notch (middle), and the end. Each piece has an outline which gives the appearance of a single continuous outline. I previously rendered the label inside of the notch so that the "cut out" with stacked/floating labels appears, but that caused the notch to be the wrong size.

The middle piece has always been needed with `fill="outline"` for all labels, but creating the cut out by removing the top border is only used by stacked/floating labels.

The 3 pieces are only rendered if `fill==="outline"` on MD mode (see `hasOutlineFill` above)